lib/paquito/types.rb

```diff
@@ -178,21 +220,42 @@ def unpackers
           time_zone = ::Time.find_zone(time_zone_name)
           ActiveSupport::TimeWithZone.new(time, time_zone)
         end,
-      },
-      "Set" => {
+      }.freeze,
+      {
         code: 9,
+        class: "Set",
+        version: 0,
         packer: ->(value, packer) { packer.write(value.to_a) },
         unpacker: ->(unpacker) { unpacker.read.to_set },
         recursive: true,
-      },
-      # Integer => { code: 10 }, reserved for oversized Integer
-      # Object => { code: 127 }, reserved for serializable Object type
-    }
+      }.freeze,
+      # { code: 10, class: "Integer" }, reserved for oversized Integer
+      {
+        code: 11,
+        class: "Time",
+        version: 1,
+        # Unfortunately there is no way to instantiate a Time with a #zone
+        # so we have to fallback to Marshal.
```

Comment:
I'm not against the change, but this feels like a deficiency in Ruby. Could we expose this somehow (in Ruby) so we wouldn't need to resort to Marshal in the future?

Reply:
@shioyama FYI I'll close this one in favor of #29 when I get back to work next week, but I need to modify it a bit as we can actually skip the rational object and just store two integers. (ref: https://bugs.ruby-lang.org/issues/19253)